target/linux/mediatek/dts/mt7981b-openwrt-one.dts

```diff
@@ -346,6 +346,14 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&spi1_flash_pins>;
 	status = "okay";
+        spidev@0 {
+                compatible = "linux,spidev";
```

Comment:
Is this working https://elixir.bootlin.com/linux/v6.12.4/source/drivers/spi/spidev.c#L727 ?

Reply:
yes, the comment above applies ?
/*
 * spidev should never be referenced in DT without a specific compatible string,
 * it is a Linux implementation thing rather than a description of the hardware.
 */

Reply:
AFAICT it won't match, since "linux,spidev" != "spidev" and it does an exact match (`!strcmp()`).

That being said, a generic spidev compatible has been rejected upstream multiple times, and the proper way here is to use devicetree overlays with the actual device being populated.

Reply:
Oh ok, i see. (just meshtasticd dont work when overlays are used)

Reply:
i hope to re-submit once i have it implemented with device tree overlay

Reply:
> AFAICT it won't match, since "linux,spidev" != "spidev" and it does an exact match (!strcmp()).

Ah, it was changed, nice. I just remember patching it out like we've it currently in RPI

https://github.com/openwrt/openwrt/blob/1375a7bae0d5e8dddb6b6cfe0584a399cd6c20b4/target/linux/bcm27xx/patches-6.6/950-1232-spi-spidev-Restore-loading-from-Device-Tree.patch#L22